std/path/posix.ts

```diff
@@ -24,26 +24,16 @@ export function resolve(...pathSegments: string[]): string {
   let resolvedPath = "";
   let resolvedAbsolute = false;
 
-  for (let i = pathSegments.length - 1; i >= -1 && !resolvedAbsolute; i--) {
-    let path: string;
-
-    if (i >= 0) path = pathSegments[i];
-    else {
-      if (globalThis.Deno == null) {
-        throw new TypeError("Resolved a relative path without a CWD.");
```

Comment:
I think these errors should stay. I think otherwise it is possible that we return a relative path from resolve.

It would be great if you added a test to check what happens when you only pass relative path segments to resolve.

```
    if (i >= 0) path = pathSegments[i];	
    else {	
        throw new TypeError("Resolved a relative path without a CWD.");	
    }	
```

Reply:
Yeah, makes sense. I had a question though. Is there a deno API (rutime or std) that would convert relative path to absolute path other than path.resolve? If not, should we provide one as we are removing that part from path.resolve.

Reply:
cc @nayeemrmn

I think we should introduce `fs.resolvePath`, which does the same as current `path.resolve` (so fall back to Deno.cwd). I think join and resolve have different semantics. I dont think we can join them:
```js
path.join("./foo/", "/bar",) // "./foo/bar"
path.resolve("./foo/", "/bar",) // "/bar"
```

Reply:
Yes, exactly. It will be a cascade of changes if we go ahead with changing ```path.resolve's``` implementation. The same can be done for path.relative by introducing a similar function under fs.